rotkehlchen/tasks/assets.py

```diff
@@ -74,3 +108,116 @@ def autodetect_spam_assets_in_db(user_db: DBHandler) -> None:
                 'blockchain': chain.to_blockchain().serialize(),
             },
         )
+
+
+def augmented_spam_detection(user_db: DBHandler) -> None:
+    """
+    Make a more exhaustive check looking for spam tokens. What it does is:
+    1. Query the assets in the history events
+    2. Filter the assets that have only receive events
+    3. Check if the logs for the tx receipt has more than MULTISEND_SPAM_THRESHOLD transfers
+    4. Check price using defillama as oracle
+
+    For any asset that passes all the previous filters we mark it as spam and it is ignored.
+    """
+    history_db = DBHistoryEvents(user_db)
+    evm_tx_db = DBEvmTx(user_db)
+    inquirer = Inquirer()
+    globaldb = GlobalDBHandler()
+    usd_asset = A_USD.resolve_to_fiat_asset()
+    chains_with_spam: set[ChainID] = set()
+
+    spam_assets: set[str] = set()
+    with user_db.conn.read_ctx() as cursor:
+        cursor.execute('SELECT DISTINCT asset FROM history_events')
+        for (asset_id,) in cursor:
+            asset = Asset(asset_id)
+            if (
+                not asset.is_evm_token() or
+                asset.identifier in spam_assets
+            ):
+                continue
+
+            events: list['EvmEvent'] = history_db.get_history_events(
+                cursor=cursor,
+                filter_query=EvmEventFilterQuery.make(assets=(asset,)),
```

Comment:
Be explicit here. Even if it is now the default, you have to be explicit ... and I think it's not the default and this will return the latest event first.

From what I understand you want to return the events, earliest first. Also please add it in a compact comment.

```suggestion
                filter_query=EvmEventFilterQuery.make(assets=(asset,)),
```